core/primitives/src/sharding.rs

```diff
@@ -1128,7 +1110,7 @@ impl EncodedShardChunk {
         #[cfg(feature = "protocol_feature_block_header_v3")]
         let block_header_v3_version = Some(ProtocolFeature::BlockHeaderV3.protocol_version());
 
-        if protocol_version < SHARD_CHUNK_HEADER_UPGRADE_VERSION {
+        let (new_chunk, merkle_paths) = if protocol_version < SHARD_CHUNK_HEADER_UPGRADE_VERSION {
```

Comment:
It seems that you pattern match here only to return the same value. What is the point of doing this?

Reply:
Yeah, I missed it while moving this part of code.